extensions/amp-viafoura/0.1/amp-viafoura.js

```diff
@@ -38,40 +40,64 @@ class AmpViafoura extends AMP.BaseElement {
     return isLayoutSizeDefined(layout);
   }
 
+  getDataAttrs() {
+    return Array.prototype.slice.call(this.element.attributes)
+      .map(attr => attr.nodeName)
+      .filter(attrName => attrName.match('data-.*'))
+      .map(name => [name, this.element.getAttribute(name)])
+  }
+
+  getPageAttrs() {
+    var attributes = locations();
+    attributes.push(['url', location.href])
+    return loc;
+  }
+
+  getParams() {
+    return this.getDataAttrs()
+      .concat(this.getPageAttrs())
+      .map(pair => `${pair[0]}=${encodeURIComponent(pair[1])}`)
+      .reduce((prev, cur) => prev ? `${prev}&${cur}` : cur);
+  }
+
   /** @override */
   layoutCallback() {
-    const getAttr = attr => encodeURIComponent(this.element.getAttribute(attr));
-    const widget = getAttr("data-widget");
-    const limit = getAttr("data-limit");
-    const sort = getAttr("data-sort");
-    const path = getAttr("data-path");
-    const title = getAttr("data-title");
-    const unique_id = getAttr("data-unique-id");
 
     const width = this.element.getAttribute('width');
     const height = this.element.getAttribute('height');
 
     const iframe = document.createElement('iframe');
     iframe.setAttribute('frameborder', '0');
-    iframe.src = `${FRAME_SRC}?limit=${limit}&sort=${sort}&path=${path}&title=`+
-        `${title}&unique_id=${unique_id}&widget=${widget}`;
+    iframe.src = `${FRAME_SRC}?${this.getParams()}`;
 
     this.applyFillContent(iframe);
 
     iframe.width = width;
     iframe.height = height;
+    iframe.setAttribute('scrolling', 'no');
```

Comment:
An accessibility note on setting `scrolling` to `no` on iFrames: "Because many users enlarge fonts and other page elements to increase visibility and accessibility, you should not disable scrolling for iframes (or frames, for that matter) by using scrolling="no". The default scrolling value (auto) is usually the best option. You should also, when possible, design the iframe with relative sizes so the iframe element itself will scale as the page and its contents are resized." http://webaim.org/techniques/frames/